calcom/cal.com
# fix: uncaught PaymentService typings in payment utils

## What does this PR do?

Follow up to #15929

Replaced `findFirst` with `findFirstOrThrow` for better error handling in the booking process. Added additional checks for the presence of `PaymentApp` and its properties to handle missing payment services gracefully.

### What changed?
- Replaced `findFirst` with `findFirstOrThrow` for booking
- Added the `PaymentApp` type import
- Restructured `paymentApp` initialization to ensure proper type casting and null handling
- Added checks to ensure `paymentApp`, `paymentApp.lib`, and `paymentApp.lib.PaymentService` exist before proceeding
- Removed redundant booking existence check

### How to test?
1. Trigger a payment process with an existing booking and observe that it behaves as expected
2. Attempt to trigger a payment process with a non-existent booking and verify that an appropriate error is thrown
3. Ensure that the payment process correctly handles cases where the payment service is missing or improperly configured

---

## Mandatory Tasks (DO NOT REMOVE)


## Checklist



- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings
